x-pack/plugin/identity-provider/src/test/java/org/elasticsearch/xpack/idp/saml/test/IdentityProviderIntegTestCase.java

```diff
@@ -0,0 +1,273 @@
+            .put(IDP_ORGANIZATION_DISPLAY_NAME.getKey(), "Identity Provider")
+            .put(IDP_ORGANIZATION_URL.getKey(), "https://idp.org")
+            .put(IDP_CONTACT_GIVEN_NAME.getKey(), "Tony")
+            .put(IDP_CONTACT_SURNAME.getKey(), "Stark")
+            .put(IDP_CONTACT_EMAIL.getKey(), "tony@starkindustries.com")
+            .put(APPLICATION_NAME_SETTING.getKey(), "elastic-cloud")
+            .put(NAMEID_FORMAT_SETTING.getKey(), TRANSIENT)
+            .put("xpack.idp.signing.key", resolveResourcePath("/keypair/keypair_RSA_2048.key"))
+            .put("xpack.idp.signing.certificate", resolveResourcePath("/keypair/keypair_RSA_2048.crt"))
+            .put("xpack.security.authc.realms." + FileRealmSettings.TYPE + ".file.order", 0)
+            .put("xpack.security.authc.realms." + NativeRealmSettings.TYPE + ".index.order", "1")
+            .put(XPackSettings.API_KEY_SERVICE_ENABLED_SETTING.getKey(), true)
+            .put("xpack.license.self_generated.type", "trial");
+        return builder.build();
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return List.of(LocalStateIdentityProvider.class, Netty4Plugin.class, CommonAnalysisPlugin.class);
+    }
+
+    @Override
+    protected boolean addMockTransportService() {
+        return false; // security has its own transport service
+    }
+
+    @Override
+    protected boolean addMockHttpTransport() {
+        return false; // enable http
+    }
+
+    @Override
+    protected Function<Client, Client> getClientWrapper() {
+        Map<String, String> headers = Collections.singletonMap("Authorization",
+            basicAuthHeaderValue(TEST_USER_NAME, new SecureString(TEST_PASSWORD.toCharArray())));
+        // we need to wrap node clients because we do not specify a user for nodes and all requests will use the system
+        // user. This is ok for internal n2n stuff but the test framework does other things like wiping indices, repositories, etc
+        // that the system user cannot do. so we wrap the node client with a user that can do these things since the client() calls
+        // return a node client
+        return client -> (client instanceof NodeClient) ? client.filterWithHeader(headers) : client;
+    }
+
+    @Override
+    protected Path nodeConfigPath(int nodeOrdinal) {
+        return nodePath(PARENT_DIR, nodeOrdinal).resolve("config");
+    }
+
+    private String configRoles() {
+        // test role allows for everything
+        return TEST_ROLE + ":\n" +
+            "  cluster: [ ALL ]\n" +
+            "  indices:\n" +
+            "    - names: '*'\n" +
+            "      allow_restricted_indices: true\n" +
+            "      privileges: [ ALL ]\n" +
+            "  applications:\n " +
+            "    - application: elastic-cloud\n" +
+            "       resources: [ '" + SP_ENTITY_ID + "' ]\n" +
+            "       privileges: [ role:superuser ]\n";
```

Comment:
```suggestion
            "       privileges: [ sso:superuser ]\n";
```
We're proposing to name these "sso:" so it's clear within the security cluster that they have a "magic" SSO purpose and aren't just "cloud RBAC".